common/lib/src/realm_types.dart

```diff
@@ -104,35 +103,60 @@ class RealmStateError extends StateError implements RealmError {
 class Decimal128 {} // TODO Support decimal128 datatype https://github.com/realm/realm-dart/issues/725
 
 /// @nodoc
-class RealmObjectBaseMarker {}
-
-/// @nodoc
-class RealmObjectMarker extends RealmObjectBaseMarker {}
-
-/// @nodoc
-class EmbeddedObjectMarker extends RealmObjectBaseMarker {}
+abstract class RealmObjectBaseMarker {}
 
 // Union type
 /// @nodoc
-class RealmAny {
-  final dynamic value;
+class RealmValue {
```

Comment:
should we have an API doc here?

Reply:
Fixed for class. Will add for props as well